git/provider_gitlab.go

```diff
@@ -56,15 +56,37 @@ func (p *GitLabProvider) CreateRepository(ctx context.Context, r *Repository) (b
 		return false, fmt.Errorf("client error: %w", err)
 	}
 
-	var id *int
+	var gid *int
+	var projects []*gitlab.Project
 	if !p.IsPersonal {
-		groups, _, err := gl.Groups.ListGroups(&gitlab.ListGroupsOptions{Search: gitlab.String(r.Owner)}, gitlab.WithContext(ctx))
+		lgo := &gitlab.ListGroupsOptions{
+			Search:         gitlab.String(r.Owner),
+			MinAccessLevel: gitlab.AccessLevel(gitlab.GuestPermissions),
+		}
+		groups, _, err := gl.Groups.ListGroups(lgo, gitlab.WithContext(ctx))
 		if err != nil {
 			return false, fmt.Errorf("failed to list groups, error: %w", err)
 		}
 
 		if len(groups) > 0 {
-			id = &groups[0].ID
+			gid = &groups[0].ID
+		}
+
+		lpo := &gitlab.ListGroupProjectsOptions{
+			Search: gitlab.String(r.Name),
+		}
+		projects, _, err = gl.Groups.ListGroupProjects(*gid, lpo, gitlab.WithContext(ctx))
+		if err != nil {
+			return false, fmt.Errorf("failed to list projects, error: %w", err)
+		}
+	} else {
+		lpo := &gitlab.ListProjectsOptions{
+			Search: gitlab.String(r.Name),
+			Owned:  gitlab.Bool(true),
```

Comment:
Note: This ensures the listing includes only repositories owned by the current user. Lacking this was the cause of the original user-reported issue.